pagopa/io-app
# chore: [IAI-265] Refactor `IOBadge` with new variants · Remove the NativeBase `Badge`

### ⚠️ This PR depends on #4321 

## Short description
This PR updates the `IOBadge` component with new props, variants and support for dynamic font size values (applied at OS level by the user). The previous iteration only accepted the `labelColor` prop; the new one accepts a combination of the following props: `variant` (`outline` or `solid`) and `color` (color values depending on the variant prop). This PR also completely replaces all instances of the NativeBase `Badge` with the corresponding variant of the `IOBadge` component.

### Preview
<img src="https://user-images.githubusercontent.com/1255491/215811882-33a2d498-a3a1-4d23-a748-74c9fd2159d7.png" width="300" />

The component is scaling according to the `fontScale` value:

https://user-images.githubusercontent.com/1255491/216576585-cfb00bc7-daae-4382-9e5e-16a8b4827b96.mov



### Code
Before:
```js
<IOBadge text={"Badge"} small={true} labelColor={"white"} />
```
After:
```js
<IOBadge text={"Badge"} variant="solid" color="blue" small />
```

## List of changes proposed in this pull request
- Replace the former `labelColor` prop with a combination of `variant` and `color` props
- Remove the NativeBase `Badge` component from both `IOBadge` and other components
- Add the `DiscountValueBox` (used in the CGN section) to the DS `Badges` page
- Add `allowFontScaling` as prop (currently disabled) to avoid layout breaking changes

## How to test
Open the app and check out the badges in the different areas (wallet, cards, bonuses, etc…)
